### PR TITLE
[deployment_manager] Add deployment_manager

### PR DESCRIPTION
**Pull request by BenjaminPelletier**
This PR introduces a tool to automate system deployment and maintenance activities.  The current approach is to have a long, plain-English description of a set of steps a human should perform to deploy an instance of the DSS, along with other sets of instructions describing how to perform various actions on that system like forming a DSS pool.  The tool in this PR is designed to automate as much of this work as practical by performing named actions on a deployment defined declaratively.  Relative to the system we have today, the deployment definition will replace main.jsonnet, and there will be an action in deployment_manager that will correspond with calling `tk apply ...`.  deployment_manager will also be capable of performing actions that Tanka cannot, however, such as creating IP addresses, setting DNS routes, etc.  This increased capability is because Tanka only interacts with Kubernetes and deployment_manager will interact with Kubernetes plus most other well-defined things a human might do.

See deployment_manager/README.md for more information.

### CODE REVIEW

File: monitoring/deployment_manager/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+* [Form or maintain a DSS pool](../../build/pooling.md)
+* [Migrate database schemas](../../build/README.md#upgrading-database-schemas)
+* [Upgrade CockroachDB nodes](https://www.cockroachlabs.com/docs/stable/upgrade-cockroach-version.html)
+* Deploy a set of diagnostic tools ([tracer](../tracer), [uss_qualifier](../uss_qualifier), [mock_uss](../mock_uss))
+
+deployment_manager actions all accept a declarative definition of the InterUSS deployment desired.  To make changes to a deployment, the administrator should edit the development definition and then execute the desired action.
+
+## Usage
+
+To use deployment_manager, first define an InterUSS deployment in a JSON file according to the [DeploymentSpec](deployments.py) schema.  From a context with kubectl already configured to work with the target Kubernetes cluster, run [deployment_manager.py](deployment_manager.py) according to its help (`python3 deployment_manager.py --help`).
```
Comment (barroco) [inline monitoring/deployment_manager/README.md:16]: Could you please add the few steps required to install dependencies and start the python3 script ?
I tried without success the following commands:
```
$ pip3 install -r requirements.txt
$ python3 deployment_manager.py --help
Traceback (most recent call last):
  File "[REDACTED]/dss/monitoring/deployment_manager/deployment_manager.py", line 10, in <module>
    from monitoring.monitorlib.typing import ImplicitDict
ModuleNotFoundError: No module named 'monitoring'
```

Comment (barroco) [reply monitoring/deployment_manager/README.md:16]: Working steps:
```
cd monitoring/deployment_manager
export PYTHONPATH=$(pwd)/../..
pip3 install -r requirements.txt
pip3 install -r ../monitorlib/requirements.txt
python3 deployment_manager.py --help
```

File: monitoring/deployment_manager/actions/dss/v1/info.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def _public_key_bytes(public_key) -> bytes:
+    return public_key.public_bytes(
+        cryptography.hazmat.primitives.serialization.Encoding.PEM,
+        cryptography.hazmat.primitives.serialization.PublicFormat.SubjectPublicKeyInfo)
+
+
+@deployment_action('dss/info/print_ca_public_certs')
+def print_ca_public_certs(context: Context):
```
Comment (barroco) [inline monitoring/deployment_manager/actions/dss/v1/info.py:22]: This action should fail gracefully if the namespace doesn't exist. Currently the execution `python3 deployment_manager.py dss/info/print_ca_public_certs examples/test.json` fails with the following trace:
```
Traceback (most recent call last):
  File "[REDACTED]/dss/monitoring/deployment_manager/deployment_manager.py", line 59, in <module>
    sys.exit(main())
  File "[REDACTED]/dss/monitoring/deployment_manager/deployment_manager.py", line 53, in main
    action_method(context)
  File "[REDACTED]/dss/monitoring/deployment_manager/actions/dss/v1/info.py", line 26, in print_ca_public_certs
    ca_crt: V1Secret = context.clients.core.read_namespaced_secret(name='cockroachdb.ca.crt', namespace=context.spec.dss.v1.namespace)
  File "[REDACTED]/dss/monitoring/monitorlib/typing.py", line 156, in __getattribute__
    return self[item]
KeyError: 'dss'
```

**Review by barroco — APPROVED** [review]
@BenjaminPelletier, thank you for this PR. The tool looks very promising. You will find some suggestion found while running the tool in comments.

File: monitoring/deployment_manager/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+If minikube is already installed, then start a session with `minikube start`.
+
+If kubectl is not currently configured to control minikube: `kubectl config use-context minikube`
+
+View the minikube dashboard with `minikube dashboard`.
+
+### Hello world
+
+The actions in [test/hello_world](actions/text/hello_world.py) demonstrate manipulation of [a simple system](actions/test/README.md) using deployment_manager; see [the README](actions/test/README.md) for actions that can be performed on this system.
```
[the remark this answers is not in the document]
Comment (BenjaminPelletier) [reply monitoring/deployment_manager/README.md:37]: Nice catch

File: monitoring/deployment_manager/README.md (hunk 1 above)
Comment (BenjaminPelletier) [reply monitoring/deployment_manager/README.md:16]: Done

File: monitoring/deployment_manager/actions/dss/v1/info.py (hunk 2 above)
Comment (BenjaminPelletier) [reply monitoring/deployment_manager/actions/dss/v1/info.py:22]: Done

**Pull request merged**